tests/tensorflow2/test_keras.py

```diff
@@ -588,6 +588,33 @@ def test_layer_names(out_dir, tf_eager_mode):
         assert re.match(pattern=pattern, string=tname) is not None
 
 
+@pytest.mark.slow
+def test_regex_filtering_for_default_collections(out_dir):
+    hook = smd.KerasHook(
+        out_dir,
+        save_config=SaveConfig(save_interval=9),
+        include_collections=[CollectionKeys.LAYERS, CollectionKeys.GRADIENTS],
+    )
+    hook.get_collection("layers").include("^dense")
```

Comment:
If this is how we specify regex for individual collections, then what does the `include_regex` parameter in the hook do? What happens when `include_regex` is specified with `include_collections`?

Reply:
Customers can do both.

If they use the parameter at the hook level like this:

```
hook = smd.KerasHook(
include_regex=
    )
```

The filter is applied to all collections.
The example in the code, applies the filter to that particular collection only.